### PR TITLE
Upgrade stack config module w/ support for workspace & component inheritance

### PR DESCRIPTION
**Pull request by danjbh**
## what & why?

* Bumping version of terraform-yaml-stack-config to support component inheritance
* Updating environment module to support new spacelift `settings` section generated by terraform-yaml-stack-config
* Adding support for `terraform_workspace` in Spacelift provider (`workspace` is also an output of `terraform-yaml-stack-config`)
* Updating GIthub workflows & context.tf in a couple spots

### CODE REVIEW

File: modules/stack/main.tf [hunk 1]
```diff
@@ -26,7 +27,7 @@ resource "spacelift_mounted_file" "stack_config" {
     for k, v in var.component_vars : k => jsondecode(v)
   }))
 
-  write_only = true
+  write_only = false
```
Comment (danjbh) [inline modules/stack/main.tf:30]: I'm setting this permanently as it doesn't contain anything sensitive and is frequently useful for troubleshooting stack config issues.

File: context.tf [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 module "this" {
   source  = "cloudposse/label/null"
-  version = "0.22.1" // requires Terraform >= 0.12.26
+  version = "0.23.0" // requires Terraform >= 0.13.0
```
Comment (aknysh) [inline context.tf:23]:
```suggestion
  version = "0.24.1" // requires Terraform >= 0.13.0
```

File: examples/complete/context.tf [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+# and access individual variables as `module.this.<var>`,
+# with final values filled in.
+#
+# For example, when using defaults, `module.this.context.delimiter`
+# will be null, and `module.this.delimiter` will be `-` (hyphen).
+#
+
+module "this" {
+  source  = "cloudposse/label/null"
+  version = "0.23.0" // requires Terraform >= 0.13.0
```
Comment (aknysh) [inline examples/complete/context.tf:23]:
```suggestion
  version = "0.24.1" // requires Terraform >= 0.13.0
```

**Review by osterman — APPROVED** [review]

**Review by aknysh — APPROVED** [review]

**Pull request merged**